zcash_client_backend/src/keys.rs

```diff
@@ -74,12 +86,56 @@ pub enum DerivationError {
     Transparent(hdwallet::error::Error),
 }
 
+/// A version identifier for the encoding of unified spending keys.
+///
+/// Each epoch corresponds to a range of block heights. During an epoch, the unified spending key
+/// parsed from an encoded form tagged with that epoch's identifier is expected to provide
+/// sufficient spending authority to spend any non-Sprout shielded note created in a transaction
+/// within the epoch's block range.
+#[derive(Debug, PartialEq, Eq)]
+pub enum Epoch {
+    /// The Orchard epoch begins at Orchard activation, and will end if a new pool that requires a
+    /// change to unified spending keys is introduced.
+    Orchard,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum DecodingError {
+    ReadError(&'static str),
+    EpochInvalid,
+    EpochMismatch(Epoch),
```

Comment:
Don't forget to change these (e.g. to `EraInvalid` and `EraMismatch(Era)`) as well.